programs/bubblegum/program/src/lib.rs

```diff
@@ -251,6 +251,64 @@ pub struct Delegate<'info> {
     pub system_program: Program<'info, System>,
 }
 
+#[derive(Accounts)]
+pub struct UpdateMetadata<'info> {
+    /// CHECK: Can optionally specify the old_metadata of the leaf through an account to save transaction space
+    /// CHECK: This account is checked in the instruction
+    pub old_metadata_acct: Option<UncheckedAccount<'info>>,
+    #[account(
+        seeds = [merkle_tree.key().as_ref()],
+        bump,
+    )]
+    /// CHECK: This account is neither written to nor read from.
+    pub tree_authority: Account<'info, TreeConfig>,
+    pub tree_delegate: Signer<'info>,
+    /// CHECK: This account is checked in the instruction
+    pub leaf_owner: UncheckedAccount<'info>,
+    /// CHECK: This account is chekced in the instruction
+    pub leaf_delegate: UncheckedAccount<'info>,
+    pub payer: Signer<'info>,
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program
+    pub merkle_tree: UncheckedAccount<'info>,
+    pub log_wrapper: Program<'info, Noop>,
+    pub compression_program: Program<'info, SplAccountCompression>,
+    pub token_metadata_program: Program<'info, MplTokenMetadata>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct UpdateMetadataCollectionNFT<'info> {
```

Comment:
I am not sure we need 2 separate `Update` instructions.

Reply:
Will remove ``update_metadata_collection_nft` from our branch as follow-on PR.